internal/access/credential.go

```diff
@@ -184,35 +184,68 @@ func hasMinimumCount(min int, password string, check func(rune) bool) bool {
 	return count >= min
 }
 
+func passwordRequirements(s *models.Settings) []string {
+	requirements := make([]string, 0)
+
+	format := func(n int, single, multiple string) string {
+		if n > 1 {
+			return fmt.Sprintf(multiple, n)
+		}
+
+		return fmt.Sprintf(single, n)
+	}
+
+	if s.LengthMin > 0 {
```

Comment:
This feels like it would be easier to read if you consolidated it w/ `checkPasswordRequirements()` and used a struct/slice to hold each of the requirements/format strings.

Reply:
```
type ReqCheck func(rune) bool

type PasswordReq struct {
        Min      int
        Singular string
        Plural   string
        Check    ReqCheck
}

func checkPasswordRequirements(db data.ReadTxn, password string) error {
	settings, err := data.GetSettings(db)
	if err != nil {
		return err
	}
	
        reqs := []PasswordReq{
                {s.LowercaseMin, "%d lowercase letter", "%d lowercase letters", unicode.IsLower},
                {s.UppercaseMin, "%d uppercase letter", "%d uppercase letters", unicode.IsUpper},
                {s.NumberMin, "%d number", "%d numbers", unicode.IsDigit},
                {s.SymbolMin, "%d symbol", "%d symbols", isSymbol},
        }
        reqString := []string{}

        for _, r := range reqs {
                if !hasMinimumCount(r.Min, password, r.Check) {
                        if r.Min > 1 {
                                reqString = append(reqString, fmt.Sprintf(r.Plural, r.Min))
                        } else {
                                reqString = append(reqString, fmt.Sprintf(r.Singular, r.Min))
                        }
                }
        }
        ...
        
}
```
You would need to add the test back for the minimum length as well. You can either add it in `reqs` (but not pass it to `hasMinimumCount()`), or just do it one off. Then just pass the error back if there is one.

Reply:
This snippet has the downside where only the unmet requirements are added to the error, one of the problems this PR is trying to address but I get the idea